graph/patterns/alternate-key.md

```diff
@@ -0,0 +1,48 @@
+# Alternate Key Pattern
+
+Microsoft Graph API Design Pattern
+
+*The Alternate Key Pattern provides the ability to query for a single, specific entity identifiable through an alternative attribute (called key) that is not its unique identifier*
+
+## Problem
+--------
+
+All entities in our system are identified by an UUID - which guarantees uniqueness. Often though, that same entity can also be uniquely identified by an alternative, more convenient attribute.
+
+Take a look at the `user` entity: while the UUID remains a perfectly valid way to get the entity details, the `email` address is also an unique attribute that could be used to identify it.
```

Comment:
If we are using the Microsoft Graph `user` resource type as an example, we should use its property names:
UUID -> `id`
`email` -> `mail`  (this one is unfortunate, as this property really should have been called email :( )